src/Repository/OrderRepository.php

```diff
@@ -55,7 +55,15 @@ public function createFromCart(Cart $cart, string $orderNumber = null): Order
             throw APIException::fromResponse($response);
         }
 
-        $documentManager->getUnitOfWork()->registerAsManaged($order, $order->getId(), $order->getVersion());
+        /** @var Cart $cart */
+        $cart = $this->documentManager->getRepository(get_class($cart))->findAndCreateObject($cart->getId(), false);
```

Comment:
I didn't want to create a direct dependency between the repositories to ease up the declaration process, any objections?

Reply:
Maybe instead of using the cart repository just fire a raw request, like you do for the order already?